### ISSUE TITLE
[Bug] Issue in reversing graph that was loaded from file

### ISSUE THREAD

**Issue by JakeStevens** [dmlc/dgl#1081]
## 🐛 Bug

Reversing a graph that was loaded from a file using shared_*data does not result in a correct reversed graph for certain graph topologies. The graph's edges are mutated on the first reverse, resulting in the incorrect graph. Subsequent calls to reverse do not mutate the graph further.

## To Reproduce

```
from dgl import DGLGraph
from dgl.data.utils import save_graphs, load_graphs
import torch

graph = DGLGraph()
graph.add_nodes(3)
graph.add_edge(1, 0)
graph.add_edge(0, 1)
graph.add_edge(2, 0)
graph.edata['e_feats'] = torch.Tensor([[0, 1, 0],
                              [1, 1, 1],
                              [1, 0, 1]])
save_graphs('./graph.bin', graph)


before_direct = graph.edges[[0,1,2],[1,0,0]].data['e_feats']
g = graph.reverse(share_ndata=True, share_edata=True)
after_direct = g.edges[[1,0,0],[0,1,2]].data['e_feats']

graphs, _ = load_graphs('./graph.bin')
graph = graphs[0]

before_load = graph.edges[[0,1,2],[1,0,0]].data['e_feats']
g = graph.reverse(share_ndata=True, share_edata=True)
after_load = g.edges[[1,0,0],[0,1,2]].data['e_feats']
g2 = g.reverse(share_ndata=True, share_edata=True)
after_load_2nd_rev = g2.edges[[0,1,2],[1,0,0]].data['e_feats']

print(torch.all(before_direct == after_direct)) #should be true, is true
print(torch.all(before_load == after_load)) #should be true, is false
print(torch.all(after_load == after_load_2nd_rev)) #should be true, is true
```

## Expected behavior

The edge data for an edge (e.g, (1 -> 0)) should be the same as the edge data for the reversed edge (e.g, (0->1). This occurs for a generated graph but not for a loaded graph.

## Environment

 - DGL Version (e.g., 1.0): latest commit
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): PyTorch 0.4.1
 - OS (e.g., Linux): Linux
 - How you installed DGL (`conda`, `pip`, source): source
 - Build command you used (if compiling from source): NOCUDA
 - Python version: 3.7
 - CUDA/cuDNN version (if applicable): N/A
 - GPU models and configuration (e.g. V100): N/A
 - Any other relevant information:

**Comment by mufeili** [comment]
Thank you for the report and I've verified the bug. @VoVAllen Could you please take a look? The implementation for `reverse` can be found [here](https://github.com/dmlc/dgl/blob/master/python/dgl/transform.py#L287).

**Comment by VoVAllen** [comment]
@mufeili 
https://github.com/dmlc/dgl/blob/master/python/dgl/transform.py#L291
`g.edges()` doesn't guarantee the edge ordering by eids. `g.all_edges(order='eid')` did. However I think we should change the behavior of `g.edges()` to force it return ordered edges. @jermainewang What do you think about this?

**Comment by mufeili** [comment]
This should be fixed in #1090 .

**Comment by JakeStevens** [comment]
Can confirm that #1090 fixes both the bug in both the small test case I shared, as well as in the larger application I was writing.
